lib/mongo/cluster/reapers/cursor_reaper.rb

```diff
@@ -195,6 +195,12 @@ def kill_cursors
             connection_global_id: kill_spec.connection_global_id,
           }
           op.execute(server, context: Operation::Context.new(options: options))
+
+          if session = kill_spec.session
+            if session.implicit? && !session.ended?
```

Comment:
Is this cleanup now happening earlier compared to when it happened before?

Reply:
I wouldn't say so. First, before it never happened because it tried to use a mutex in a finalizer. Second, this is now called in a job that is done by a periodic executor, so it is most probably delayed. And finally, it is the last operation in the killing cursor routine.

Reply:
Done